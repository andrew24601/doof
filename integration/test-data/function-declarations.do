
function greet(name: string): string {
    return "Hello " + name;
}
function add(a: int, b: int): int {
    return a + b;
}
function multiply(x: int, y: int = 2): int {
    return x * y;
}
function factorial(n: int): int {
    if (n <= 1) {
        return 1;
    }
    return n * factorial(n - 1);
}

function main(): int {
    let result = "";
    const appendToResult = (text: string): void => {
        result = result + text;
    };
    let greeting = greet("World");
    let sum = add(3, 4);
    let product1 = multiply(5, 2);
    let product2 = multiply(5, 3);
    let fact = factorial(4);
    appendToResult(greeting);
    appendToResult("|");
    appendToResult(string(sum));
    appendToResult("|");
    appendToResult(string(product1));
    appendToResult("|");
    appendToResult(string(product2));
    appendToResult("|");
    appendToResult(string(fact));
    println(result);
    return 0;
}