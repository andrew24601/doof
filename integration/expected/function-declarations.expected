Hello World|7|10|15|24